### PR TITLE
Twilight forest 1.18 port fix

### PR DESCRIPTION
**Pull request by baguchi**
I made the 1.18 twilight forest port with my tiny(or dirty) fix

Here is what I doing:
+readd TFBiomeProvider
+fixed data generator
-removed overworld feature in twilight forest(because It' cause datapack error)
+Fixed TF Caves Carver Crash(Temporary)
+fixed Structure crash
+fixed tree decorator registry

### CODE REVIEW

File: src/main/java/twilightforest/world/components/TFCavesCarver.java [hunk 1]
```diff
@@ -78,6 +77,8 @@ public boolean carve(CarvingContext ctx, CaveCarverConfiguration config, ChunkAc
 
 	@Override
 	protected boolean carveBlock(CarvingContext ctx, CaveCarverConfiguration config, ChunkAccess access, Function<BlockPos, Biome> biomePos, CarvingMask mask, BlockPos.MutableBlockPos pos, BlockPos.MutableBlockPos posUp, Aquifer aquifer, MutableBoolean isSurface) {
+		//TODO mayby another way...?
+		Random random = new Random();
```
Comment (Drullkus) [inline src/main/java/twilightforest/world/components/TFCavesCarver.java:81]: This is terrible for deterministic world generation, please see if you can get an instance of a `Random` from one of the parameters or seed it instead

**Review by GizmoTheMoonPig — APPROVED** [review]
works for me! if anything is messed up its fine, I can always fix it later

File: src/main/java/twilightforest/world/registration/TFFeature.java [hunk 2]
```diff
@@ -944,6 +944,7 @@ public StructurePiece provideFirstPiece(StructureManager structureManager, Chunk
 		return Optional.ofNullable(this.provideFirstPiece(structureManager, chunkGenerator, random, x, y, z));
 	}
 
+	//TODO Mayby better way has....?
```
Comment (Drullkus) [inline src/main/java/twilightforest/world/registration/TFFeature.java:947]: What's this comment for?

File: src/main/java/twilightforest/world/registration/TwilightFeatures.java [hunk 3]
```diff
@@ -52,10 +42,9 @@
     }
 
     private static <P extends FoliagePlacer> FoliagePlacerType<P> registerFoliage(ResourceLocation name, Codec<P> codec) {
+        //TRUNK REPLACER register forge side is current not good
```
Comment (Drullkus) [inline src/main/java/twilightforest/world/registration/TwilightFeatures.java:45]: What's not good about it?

File: src/main/java/twilightforest/world/registration/biomes/BiomeHelper.java [hunk 4]
```diff
@@ -465,10 +464,10 @@ public static void addHighlandCaves(BiomeGenerationSettings.Builder biome) {
...
         BiomeGenerationSettings.Builder biome = new BiomeGenerationSettings.Builder();
 
-        BiomeDefaultFeatures.addSwampClayDisk(biome);
-		BiomeDefaultFeatures.addDefaultSoftDisks(biome);
-        BiomeDefaultFeatures.addForestGrass(biome);
-        BiomeDefaultFeatures.addSavannaGrass(biome);
+        //BiomeDefaultFeatures.addSwampClayDisk(biome);
+		//BiomeDefaultFeatures.addDefaultSoftDisks(biome);
+        //BiomeDefaultFeatures.addForestGrass(biome);
+        //BiomeDefaultFeatures.addSavannaGrass(biome);
```
Comment (Drullkus) [inline src/main/java/twilightforest/world/registration/biomes/BiomeHelper.java:470]: Why are these commented out?

**Review by Drullkus — CHANGES_REQUESTED** [review]
I'm not comfortable accepting this PR. We're going to dispose the old Biome Source with a new one I'm writing. There appears to be a few features that got commented out without explanation as well.

There will be a `BiomeSource` overhaul and a `StructureFeature` overhaul that'll deprecate `TFFeature` as well

**Comment by Drullkus** [comment]
Wait wtf this was pulled

**Comment by Tamaized** [comment]
It was pulled so giz can work on the stuff he needs to do. such as the rest of the features

**Pull request merged**